apps/opentelemetry_api/lib/open_telemetry/tracer.ex

```diff
@@ -127,7 +127,16 @@ defmodule OpenTelemetry.Tracer do
   The Span in the current Context has its `is_recording` set to `false`.
   """
   def end_span() do
-    non_recording_span = :otel_span.end_span(:otel_tracer.current_span_ctx())
+    do_end_span(:opentelemetry:timestamp())
```

Comment:
```suggestion
    do_end_span(:opentelemetry.timestamp())
```

But preferably only in the span utils module is the timestamp created unless passed to the user. I think the couple of lines that have to be duplicated by not sharing `do_end_span` between the two functions but instead each one calling `set_current_span` is fine.

Actually, `set_current_span` returns the span now, so you don't need to have the extra line returning `non_recording_span` anymore and can get rid of `_ = `